### ISSUE TITLE
Enable Specs URLs to Return JSON

### ISSUE THREAD

**Issue by lleadbet** [apollographql/specs#22]
It would be nice if we could return a JSON blob of the directives and their definition specific to a federation version; for example: https://specs.apollo.dev/federation/v2.0?format=json
would return something like:

```json
{
    "@key":"directive @key(fields: _FieldSet!, resolvable: Boolean) repeatable on OBJECT | INTERFACE",
    "@link":"directive @link(import: [link_Import!], url: String!) repeatable on SCHEMA\nscalar link_Import",
    "@shareable":"directive @shareable on OBJECT | FIELD_DEFINITION"
}
```

As we continue to iterate on Federation 2 with new versions and had to update definitions (as we did with `@key` from a single arg to two from Fed1->2), it would be a lot easier for devs to programmatically support new versions as well as being more definitive on what those definitions should be.

Additionally, we should evaluate if we should dictate whether the directive is handled at the gateway/router level, or subgraph; if we want to fully support new directives, we should include information whether there is implementation details that the third party needs to adjust- for example, `@key` requires work by the library to implement entities and entity resolvers to support fully.

**Comment by queerviolet** [comment]
hey! so we fully intend to support a process for getting the schema from a link url. that process (sketched) is:

1. `GET $url` with `Accept: application/graphql`
2. `GET $url.graphql`
3.  treat `$url` as a graphql endpoint, do federated service discovery
4. treat `$url` as a graphql endpoint, do introspection

i've implemented it in this branch: https://deploy-preview-21--apollo-specs.netlify.app/

e.g., the federation 2.0 schema as it will deploy to [https://specs.apollo.dev/federation/v2.0.graphql](https://deploy-preview-21--apollo-specs.netlify.app/federation/v2.0.graphql)

**Comment by queerviolet** [comment]
i should ask, does that address your needs?

i'm not opposed to another format, but it opens up some issues. namely, with JSON (or some other minimally structured format we pick) we lose any directives on the schema element, so we lose any `@link`s the schema has, so if one core schema links to another, we can't represent it. your example actually demonstrates this:

```graphql
    "@link":"directive @link(import: [link_Import!], url: String!) repeatable on SCHEMA\nscalar link_Import",
```

the directive definition here contains `link__Import`. but without the schema's headers, we don't know what the `link__` prefix means, so we can't resolve the reference to a URL. automated tools trying to crawl the dependency graph are stuck.

we could of course include whatever we need in whatever response container, but it seems like then we'll have another format to maintain. i'm not clear on the value, either—you need a graphql parser to interpret this JSON anyway, so it seems like just parsing the whole response would be about as easy?

**Comment by queerviolet** [comment]
assuming graphql is an acceptable response type, i believe this is fixed by #21!

e.g. the sdl for federation/v2.0 is now available at https://specs.apollo.dev/federation/v2.0.graphql

please open another issue if you'd like another response format
